src/V3Width.cpp

```diff
@@ -2469,23 +2469,45 @@ class WidthVisitor final : public VNVisitor {
             nextip = itemp->nextp();  // iterate may cause the node to get replaced
             VL_DO_DANGLING(userIterate(itemp, WidthVP{CONTEXT_DET, PRELIM}.p()), itemp);
         }
-        // Take width as maximum across all items
-        int width = nodep->exprp()->width();
-        int mwidth = nodep->exprp()->widthMin();
-        for (const AstNode* itemp = nodep->itemsp(); itemp; itemp = itemp->nextp()) {
-            width = std::max(width, itemp->width());
-            mwidth = std::max(mwidth, itemp->widthMin());
+
+        AstBasicDType* dtype = VN_CAST(nodep->exprp()->dtypep(), BasicDType);
+        AstNodeExpr* (*eq_ctor)(FileLine*, AstNodeExpr*, AstNodeExpr*) = nullptr;
+        AstNodeDType* subDTypep = nullptr;
+
+        if (dtype && dtype->isString()) {
+            nodep->dtypeSetString();
+            eq_ctor = [](FileLine* fl, AstNodeExpr* lhsp, AstNodeExpr* rhsp) -> AstNodeExpr* {
+                return new AstEqN{fl, lhsp, rhsp};
+            };
```

Comment:
I think AstEq::newTyped does part of this for you.

Reply:
`AstEq::newTyped` defaults to `AstEq` node instead of `AstEqWild`.